### PR TITLE
Adds github job to prerelease next on push

### PR DESCRIPTION
**Pull request by matdru**
## New
- on each push to next branch, prerelease version will be published to npm
- on PR to develop and next, run code analysis workflow

### CODE REVIEW

File: .github/workflows/next.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Build
+        run: yarn build
+
+      - name: Setup git
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+
+      - name: Publish
+        run: yarn publish:beta
```
Comment (prevwong) [inline .github/workflows/next.yml:75]: @matdru I thought we were using the alpha prefix for the next branch (for now), no?

Comment (matdru) [reply .github/workflows/next.yml]: Yeaj i'll switch to conventional prerelease that will increment alphas by default

File: .github/workflows/next.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Build
+        run: yarn build
+
+      - name: Run tests
+        run: yarn test
+
+  prerelease:
+    name: Prerelease
+    if: github.ref == 'refs/heads/master'
```
Comment (prevwong) [inline .github/workflows/next.yml:40]: This doesn't seem right, I think? Wouldn't this not run the NPM release for the `next` branch?

**Review by prevwong — APPROVED** [review]

**Comment by matdru** [comment]
My bad, made a typo in job name, needs one extra commit

**Pull request merged**